src/app/services/articles.service.ts

```diff
@@ -1,66 +1,32 @@
 import { Injectable } from '@angular/core';
-import { BehaviorSubject, map, distinctUntilChanged } from "rxjs";
-import { HttpClient } from '@angular/common/http'
+import { of } from "rxjs";
+import { HttpClient, HttpHeaders } from '@angular/common/http'
 
+import { UserService } from "./user.service";
 import { ArticleModel } from "../models/article.model";
 
 @Injectable({
   providedIn: 'root'
 })
 export class ArticlesService {
-  private articles = new BehaviorSubject([]);
+  popularTags$ = of([]);
+  headers = new HttpHeaders().set('User-ID', this.userService.getId());
 
-  articles$ = this.articles.asObservable();
+  constructor(private http: HttpClient, private userService: UserService) {}
 
-  popularTags$ = this.articles$.pipe(
-    distinctUntilChanged((previous, next) => previous.length === next.length),
-    map(articles => {
-      if (!articles.length) {
-        return [];
-      }
-
-      const allTags = articles.reduce((result, article) => [...result, ...article.tagList], []);
-      const tagNameToCountMap = allTags.reduce((result, tagName) => ({
-          ...result,
-          [tagName]: result[tagName] ? result[tagName] + 1 : 1
-      }), {});
-
-      return Object.entries(tagNameToCountMap).sort(([,aCount], [, bCount]) => (bCount as number) - (aCount as number))
-        .slice(0, 3)
-        .map(([name]) => name);
-    })
-  );
-
-  constructor(private http: HttpClient) {}
-
-  getArticles() {
-    return this.http.get<ArticleModel[]>('/api/posts');
+  addArticle(articleDetails: Partial<ArticleModel>) {
+    return this.http.post<ArticleModel>('/api/articles', articleDetails, {
```

Comment:
better to use full url:
``return this.http.post<ArticleModel>(`{DOMAIN_URL}/api/articles`);``

``DOMAIN_URL = 'https://localhost:3000'``